src/main/java/com/nike/cerberus/util/AwsIamRoleArnParser.java

```diff
@@ -92,7 +126,7 @@ public String convertPrincipalArnToRoleArn(final String principalArn) {
         }
 
         final boolean isAssumedRole = GENERIC_ASSUMED_ROLE_PATTERN.matcher(principalArn).find();
-        final Pattern patternToMatch = isAssumedRole ? IAM_ASSUMED_ROLE_ARN_PATTERN : IAM_PRINCIPAL_ARN_PATTERN;
+        final Pattern patternToMatch = isAssumedRole ? IAM_ASSUMED_ROLE_ARN_PATTERN : IAM_PRINCIPAL_ARN_PATTERN_ROLE_GENERATION;
```

Comment:
This method needs to still generate role ARNs for a handful of instance-profile ARNs